src/vs/workbench/common/actionRegistry.ts

```diff
@@ -52,24 +54,40 @@ export interface IWorkbenchActionRegistry {
 	 * Returns the category for the given action or null iff none.
 	 */
 	getCategory(actionId: string): string;
+
+	/**
+	 * Returns the keywords associated with the given action or null iff none.
+	 */
+	getKeywords(actionId: string): string[];
 }
 
 class WorkbenchActionRegistry implements IWorkbenchActionRegistry {
 	private workbenchActions: collections.IStringDictionary<SyncActionDescriptor>;
 	private mapActionIdToCategory: { [id: string]: string; };
+	private mapActionIdToKeywords: { [id: string]: string[]; };
 
 	constructor() {
 		this.workbenchActions = Object.create(null);
 		this.mapActionIdToCategory = Object.create(null);
+		this.mapActionIdToKeywords = Object.create(null);
```

Comment:
will it be better to actually have a map to everything? Have map where values will be some structure, so you will not need to maintain two of them?